### PR TITLE
Issue #74 Generating a form

### LINKED ISSUE #68: Update README to Include Custom Sheet Processing Roadmap
Description: As we continue developing the BubbleScan project, it’s important to keep the README file up to date with our progress and future plans. Currently, the README lacks information about our next development phase: supporting custom sheet processing. This issue focuses on adding a section to the README that outlines the planned features for custom sheet compatibility, along with any relevant notes about its development.

Goal:
Update the project’s README file to include a clear roadmap for the upcoming custom sheet processing feature. This will help future contributors understand the direction of the project and what features are expected to be implemented.
Tasks:

Add a section titled "Upcoming Features: Custom Sheet Processing" to the README.
Provide a brief description of the planned feature and its significance to the project.
List the key development milestones for implementing custom sheet processing (e.g., research, algorithm development, frontend modifications).
Ensure the new section is formatted properly and fits with the existing structure of the README.

Success Criteria:
The README file is updated with a clear and concise section on the custom sheet processing feature.

### LINKED ISSUE #74: Generate a custom exam form
**Is your feature request related to a problem? Please describe.**
Instead of purchasing exam sheets from a third party, the user should be able to generate the sheets through our application.

**Describe the solution you'd like**
The user should have an option to generate a custom exam sheet. The generated exam sheet should have space for the students to enter their name and id. The instructor will want to provide information about the number of questions to include on the exam and the number of answer choices for the questions. Bubble Scan will generate the requested exam sheet, which the instructor can print out and use in the classroom. 

**Additional context**
We will add the processing of these custom exam sheets to the application, after we client gets a chance to review the custom generated exam sheets.

### PR DESCRIPTION
**Pull request by mylescorrea**
Fixes #74  

**What was changed?**

*Added a new button that asks user for number of questions and choices. When clicking generate, a new window pops up with the generated sheet*

**Why was it changed?**

*The user needs to be able to add their specifications when requesting custom sheets*

**How was it changed?**

*Added a CustomSheetComponent.tsx file in components folder and added the different button and options in the front end App.tsx to give the user the options*

### CODE REVIEW

**Comment by yildizahmetali** [comment]
It looks like you updated the docker document successfully.

**Comment by yildizahmetali** [comment]
It looks like you updated the docker document successfully.

**Review by yildizahmetali — DISMISSED** [review]
It looks like you updated the docker document successfully.

File: bubblescan-client/src/App.tsx [hunk 1]
```diff
@@ -35,11 +36,19 @@ function App() {
...
+
+  }
+
   return (
     <div className="welcome">
       <h1>Welcome to Bubble Scan</h1>
       <h4>You can upload your files below</h4>
       <FileUploadComponent />
+      <h1>You can create Custom Sheets here</h1>
+      <button onClick={handleButtonClick}>Custom Sheet</button>
```
Comment (yildizahmetali) [inline bubblescan-client/src/App.tsx:112]: Good organization!

**Review by yildizahmetali — DISMISSED** [review]
Nice job!

**Comment by kate-holdener** [comment]
@mylescorrea Good start. Please see the details of issue #74 for what remains to be done. Keep in mind that we want to get the issue resolved in week 1 and use week 2 for review and adjustments (so the core of the work needs to be done in week 1)

**Review by Chirag2x — COMMENTED** [review]
The Front End addition for creating a custom sheet is a good progress, it also matches with the pre existing design. The code is structured well, but you should have to create a separate react component for this functionality which handles all the logic. Well since for now it's just front end part, we can rectify it in later sprint. Good work for adding the steps in future direction. Keep it up!

**Comment by kate-holdener** [comment]
@mylescorrea @Chirag2x Please make sure that this PR includes the code needed to produce the custom exam sheet (as described in issue #74). This will give you an opportunity to discuss what the form looks like with the client. The expectation is: the user enters the necessary information, clicks a button, and gets a custom exam sheet. This can all be done in the front end.

**Comment by kate-holdener** [comment]
@mylescorrea If you need a starting point for generating a custom exam sheet, don't hesitate to use AI. Here's a prompt I used and got a pretty good starting point:
```
I have a project that uses machine vision to grade scantron based exams (bubble sheets that students fill out). It works for a specific scantron sheet that instructors order. I also want to allow instructors to generate and print out their own sheets (and I'll make it work for these custom sheets). The project uses typescript (tsx). Give me the code for generating custom exam sheets. The sheets need to have space to enter student name and id, the way standard scantron sheets do. The instructor should be able to specify the number of exam questions and the number of answer options. Answer options are usually labelled with letters: A, B, C, etc
```
I used calude.ai.

**Comment by kate-holdener** [comment]
@mylescorrea, I think you'll need to tweak the generated sheet. Right now, the generated sheet has input fields for name and student id, and clickable radio buttons for bubble options. That would work well if this sheet was used on a computer based exam. However, these will be printed out, so I think the client would want to have larger circles and more of a dedicated space to write in the name and the id. You'll probably want a larger space for name and id (imagine some student with a really long name). Imagine you are a student taking this exam (on paper).

Here's the screenshot that I got from running this code.

Screenshot 2024-09-19 at 8 30 45 PM

**Comment by kate-holdener** [comment]
Once the exam sheet is generated, it would be good to get it in a pdf format too. I know you can print a page to pdf, but there may be scaling issues for different browsers, which may mess with the AI component that reads those sheets.

**Review by Chirag2x — APPROVED** [review]
Really Good Changes. You made it!

**Pull request merged**